common/FormattingUtils.js

```diff
@@ -52,10 +52,13 @@ const createFormatFunction = (web3, numDisplayedDecimals) => {
   return valInWei => formatWithMaxDecimals(formatWei(valInWei, web3), numDisplayedDecimals, false);
 };
 
-// generate a etherscan link prefix
+// generate an etherscan link prefix. If a web3 object is provided then check networkId. Else, assume mainnet.
 function createEtherscanLinkFromtx(web3) {
   // Construct etherscan link based on network
-  const networkId = web3.networkId;
+  let networkId = 1;
+  if (web3) {
+    networkId = web3.networkId;
```

Comment:
I know you didn't add this, but I don't think web3.networkId works anymore (not sure if it ever did).
We should replace with:
```
await web3.eth.net.getId()
```

Reply:
updated 👍

Reply:
After looking further into this I don't actually want to put `await web3.eth.net.getId()` into the formatter. This will make this function and everything downstream async. I've refactored this to rather take in the networkID and then get this from the `createEtherscanLinkMarkdown` using a promise. It's a bit nasty and I'm not overly happy with this implementation but changing everything downstream to be async is a worse solution imho.

let me know if you have any other ideas on improving this.

Reply:
That all sounds good as long as this doesn't break other uses of this formatter.